autorest/preparer_test.go

```diff
@@ -437,11 +438,76 @@ func TestWithFormDataSetsContentLength(t *testing.T) {
 		t.Fatalf("autorest: WithFormData failed with error (%v)", err)
 	}
 
+	expected := "age=42&name=Rob+Pike"
```

Comment:
also, you might not get these in this order. I've seen ordering of map keys in for loops change between golang versions.